nixos/modules/services/networking/mbim-network.nix

```diff
@@ -0,0 +1,127 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mbim-network;
+
+  # Transforms "/dev/cdc-wdm0" into "mbim-network-dev-cdc-wdm0" for file and
+  # systemd unit names.
+  cleanName = name: "mbim-network${replaceStrings [ "/" ] [ "-" ] name}";
+
+  makeSystemdUnit = dcfg: name:
+    let
+      # Config file format defined in 'man 1 mbim-network'.
+      configFile = with dcfg;
+        pkgs.writeText "${cleanName name}.conf" ''
+          APN=${apn}
+          PROXY=${if cfg.proxy then "yes" else "no"}
+          ${optionalString (apnUser != "") "APN_USER=${apnUser}"}
+          ${optionalString (apnPass != "") "APN_PASS=${apnPass}"}
+          ${optionalString (apnAuth != "") "APN_AUTH=${apnAuth}"}
+        '';
+
+      # Mimic the format of the commands run by mbim-network, including the
+      # optional use of mbim-proxy.
+      cliCmd = "${getBin pkgs.libmbim}/bin/mbimcli -d ${name}${
```

Comment:
This looks a bit oddly formatted. probably best to have a separate binding`getBin pkgs.libmbim` in a separate binding, we use it multiple times anyways.

Reply:
No problem, `nixfmt` did a few odd things but I went with it anyway. I'll update this.